queue_job/controllers/main.py

```diff
@@ -59,6 +59,7 @@ def _try_perform_job(self, env, job):
         http.request.env.cr.commit()
 
         _logger.debug('%s started', job)
+        job.lock()
```

Comment:
Since there is a small window between the commit and here where the job state could be reset, maybe we should do `job.lock(expected_state=STARTED)` and not exit if the expected state is not correct.